transport/src/test/java/io/camunda/zeebe/transport/stream/impl/TestCommunicationService.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.impl;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/** A mock implementation of ClusterCommunicationService */
+final class TestCommunicationService implements ClusterCommunicationService {
+
+  private final ConcurrentMap<String, BiFunction<MemberId, byte[], CompletableFuture<byte[]>>>
+      subscribers = new ConcurrentHashMap<>();
+  private final Map<MemberId, TestCommunicationService> cluster;
+
+  private final MemberId memberId;
+
+  TestCommunicationService(
+      final ConcurrentMap<MemberId, TestCommunicationService> cluster, final MemberId memberId) {
+    this.cluster = cluster;
+    this.memberId = memberId;
+    cluster.put(memberId, this);
+  }
+
+  @Override
+  public <M> void broadcast(
+      final String subject,
+      final M message,
+      final Function<M, byte[]> encoder,
+      final boolean reliable) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <M> void multicast(
+      final String subject,
+      final M message,
+      final Function<M, byte[]> encoder,
+      final Set<MemberId> memberIds,
+      final boolean reliable) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <M> void unicast(
+      final String subject,
+      final M message,
+      final Function<M, byte[]> encoder,
+      final MemberId memberId,
+      final boolean reliable) {
+    // Do nothing
```

Comment:
:wrench: If this is not implemented because we're not testing it yet, then let's add `throw new UnsupportedOperationException`. Otherwise I don't understand why it's implemented to do nothing :thinking:

Reply:
We are using unicast for sending remove requests. But since we are not waiting for ack, there is no need to implement the actual request handling. I will update the comment.

Reply:
If I wanted to run an integration test using this, and I wanted to test that the receiver did receive the request anyway, wouldn't I need to implement something here?

Reply:
Then, it's better to implement it. We cannot throw UnsupportedException because the method is used.